packages/msi_windows.go

```diff
@@ -0,0 +1,235 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package packages
```

Comment:
You may want to consider moving files like this (and rpm.go, etc.) to separate packages or sub-packages. Otherwise you run the risk of mixing the high level-logic in files like `packages_linux` with the low-level logic of interacting with the system utilities. This way, there's nothing preventing packages_linux from calling the private functions in this file which I'd think you'd want to prevent.

`coInitializeEx()` is an example where the dependency from high-level to low-level may be going in the wrong direction. Since it's used by both msi_windows and wua_windows, I could imagine a `windows` package that has a file for msi stuff, a file for wua, and another for shared package logic.

Reply:
packages is meant to be that library, it's called by other osconfig code when access to system utilities is required, not sure how moving to another internal package would help as if packages imports internal/packages, then all those internal functions are still accessible, the file level separation here is just for human readability.